server/src/main/java/io/crate/types/DataTypes.java

```diff
@@ -637,6 +637,15 @@ public static DataType<?> merge(DataType<?> leftType, DataType<?> rightType) {
             type = ObjectType.merge((ObjectType) leftType, (ObjectType) rightType);
         } else if (leftType.id() == ArrayType.ID && rightType.id() == ArrayType.ID) {
             type = new ArrayType<>(merge(((ArrayType<?>) leftType).innerType(), ((ArrayType<?>) rightType).innerType()));
+        } else if (leftType.id() == NumericType.ID && rightType.id() == NumericType.ID) {
+            Integer leftScale = ((NumericType) leftType).scale();
+            Integer rightScale = ((NumericType) rightType).scale();
+            Integer leftPrecision = leftType.numericPrecision();
+            Integer rightPrecision = rightType.numericPrecision();
+
+            Integer mergedScale = (leftScale == null || rightScale == null) ? null : Math.max(leftScale, rightScale);
+            Integer mergedPrecision = (leftPrecision == null || rightPrecision == null) ? null : Math.max(leftPrecision, rightPrecision);
```

Comment:
This would kinda contradict the precedence defined in https://github.com/crate/crate/blob/d8db3bc8f02f8310897d66ac13d5d5625afc86c0/server/src/main/java/io/crate/types/DataType.java#L179-L198

And if we change it there to have no precision > with precision, the change here shouldn't be necessary anymore. 
See also https://github.com/crate/crate/commit/1bd46a8359dc690fdb9061e69c8d0293d7f14269 for more context.
We need to try avoid spreading type specific logic

Reply:
I think cases like `NumericType(5,2)` and `NumericType(4,3)` shows that it is not always possible to determine the precedence. Then can we instead let any numeric types to be equal precedence?

Reply:
I don't have a concrete scenario that will be affected but it seems slightly off from the definition of precedence we have:
```
     * Type precedence ids which help to decide when a type can be cast
     * into another type without losing information (upcasting).
```

Reply:
Actually causes below to fail,
```
cr> select 1.1::numeric(5,1) = 1.11::numeric(4,2);
+------+
| true |
+------+
| TRUE |
+------+
SELECT 1 row in set (0.003 sec)
```